xen-bugtool

```diff
@@ -1057,6 +1057,11 @@ exclude those logs from the archive.
                                   'openvswitch/ovsdb-server.log.%d', 'openvswitch/ovsdb-server.log.%d.gz',
                                   'telemetry/telemetry.log.%d', 'telemetry/telemetry.log.%d.gz']]]
     sa_logs = get_recent_logs(glob.glob("/var/log/sa/sa[0-9][0-9]"), caps[CAP_XENSERVER_LOGS][VERBOSITY])
+    # XSI-1385: Add plain-text reports of sar data:
+    # - Collect the daily sar reports created by by sa2 each day at 23:53:
+    sa_logs += get_recent_logs(glob.glob("/var/log/sa/sar[0-9][0-9]"), caps[CAP_XENSERVER_LOGS][VERBOSITY])
+    # - Collect a report of the sar data of the current day so far:
+    cmd_output(CAP_HARDWARE_INFO, ['sar', '-A', '-f', '/var/log/sa/sa' + time.strftime('%d')])
     qemu_logs = get_recent_logs(glob.glob('/tmp/qemu.[0-9]*'), caps[CAP_XENSERVER_LOGS][VERBOSITY])
```

Comment:
A newline here goes a long way towards improving legibility.  So too at the start of the `sar` collection.

Also, you don't want random ticket references hidden like this in comments.  It will be meaningless in ~2w time when you've finished working on XSI-1385

So from a stylistic point of view only, I'd suggest
```python
                                  'telemetry/telemetry.log.%d', 'telemetry/telemetry.log.%d.gz']]]

    # Collect SAR data
    sa_logs  = get_recent_logs(glob.glob("/var/log/sa/sa[0-9][0-9]"), caps[CAP_XENSERVER_LOGS][VERBOSITY])
    sa_logs += get_recent_logs(glob.glob("/var/log/sa/sar[0-9][0-9]"), caps[CAP_XENSERVER_LOGS][VERBOSITY])
    cmd_output(CAP_HARDWARE_INFO, ['sar', '-A', '-f', '/var/log/sa/sa' + time.strftime('%d')])

    qemu_logs = get_recent_logs(glob.glob('/tmp/qemu.[0-9]*'), caps[CAP_XENSERVER_LOGS][VERBOSITY])
```

However, I don't seem to have any `/var/log/sa/sar*` files at all.  How are the human readable forms expected to appear?

What's the difference between `sar -A` and what you've got?  Nothing I can see, and I suspect this form might marginally more robust in the case of `/var/log` having filled up and there not being any recorded data.

Reply:
@andyhhp The human readable reports are/should be done each night at 23:53 on XS 8.3.0:
```bash
[root@xenserver-8.3.0]# grep -r /sa /etc/cron*|grep -v ':#'
/etc/cron.d/sysstat:*/10 * * * * root /usr/lib64/sa/sa1 1 1
/etc/cron.d/sysstat:53 23 * * * root /usr/lib64/sa/sa2 -A
```
If no `/var/log/sa/sar*` is present the day after the installation of the server, then the configuration must have been changed somehow.

Thanks for the comment on just using `sar -A`! Yes that's indeed better IMHO as well.
We received the call `sar -A -f /var/log/sa/sa<day>` from support, changed to `sar -A`!